app/components/validated_field_component.rb

```diff
@@ -12,8 +12,19 @@ def initialize(form:, name:, error_messages: {}, **tag_options)
 
   def error_messages
     {
-      valueMissing: t('simple_form.required.text'),
+      valueMissing: value_missing_error_message,
       **@error_messages,
     }
   end
+
+  private
+
+  def value_missing_error_message
+    case form.send(:default_input_type, name, form.send(:find_attribute_column, name), tag_options)
```

Comment:
>are these private methods?

They are 😬 Maybe I ought not rely on them, though I couldn't find a way to (easily) get at the same information via public methods.

Reply:
Bummer, but yeah skimming the source I'm not seeing another option either, bummer

Reply:
In an earlier iteration, I had just passed the `error_messages` option from the view for the custom message, but it felt like it could be nice if the component would infer based on the type of the input it anticipates to be rendering. Could always revert to that earlier approach in case this feels more brittle than it's worth.

Reply:
I feel like `simple_form` gem doesn't change very often or very rapidly, so my guess is this is fine for now